conjureplugin/extensionsprovider_test.go

```diff
@@ -0,0 +1,143 @@
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package conjureplugin
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestAddExtensionsToIrBytes(t *testing.T) {
+	for _, tc := range []struct {
+		name               string
+		inputIR            string
+		providedExtensions map[string]any
+		expected           string
+	}{
+		{
+			name:               "no extensions to add, empty IR",
+			inputIR:            `{}`,
+			providedExtensions: map[string]any{},
+			expected:           `{}`,
+		},
+		{
+			name:               "add extension to empty IR",
+			inputIR:            `{}`,
+			providedExtensions: map[string]any{"hello": "world"},
+			expected: `{
+	"extensions": {
+		"hello": "world"
+	}
+}`,
+		},
+		{
+			name:               "no extensions to add, empty IR extensions field returns original IR",
+			inputIR:            `{"extensions":{}}`,
+			providedExtensions: map[string]any{},
+			expected:           `{"extensions":{}}`,
+		},
+		{
+			name:               "no extensions to add, IR extensions already present returns original IR",
+			inputIR:            `{"extensions":{"already":"present"}}`,
+			providedExtensions: map[string]any{},
+			expected:           `{"extensions":{"already":"present"}}`,
+		},
+		{
+			name:               "add extension to existing IR extensions",
+			inputIR:            `{"extensions":{"already":"present"}}`,
+			providedExtensions: map[string]any{"new": "value"},
+			expected: `{
+	"extensions": {
+		"already": "present",
+		"new": "value"
+	}
+}`,
+		},
+		{
+			name:               "add extension to existing IR extensions; other IR fields are preserved",
+			inputIR:            `{"a":"b","c":"d","extensions":{"already":"present"}}`,
+			providedExtensions: map[string]any{"new": "value"},
+			expected: `{
+	"a": "b",
+	"c": "d",
+	"extensions": {
+		"already": "present",
+		"new": "value"
+	}
+}`,
+		},
+		{
+			name:               "overwrite existing extensions",
+			inputIR:            `{"extensions":{"value":"old"}}`,
+			providedExtensions: map[string]any{"value": "new"},
+			expected: `{
+	"extensions": {
+		"value": "new"
+	}
+}`,
+		},
+	} {
+		t.Run(tc.name, func(t *testing.T) {
+			actual, err := addExtensionsToIrBytes(
+				[]byte(tc.inputIR),
+				func(_ []byte, _, _, _ string) (map[string]any, error) {
+					return tc.providedExtensions, nil
+				},
+				"",
+				"",
+				"",
+			)
+			require.NoError(t, err)
+			assert.Equal(t, tc.expected, string(actual))
+		})
+	}
+	for _, tc := range []struct {
+		name               string
+		inputIR            string
+		providedExtensions map[string]any
+		wantErr            string
+	}{
+		{
+			name:    "invalid input IR: empty string",
+			inputIR: ``,
+			wantErr: `EOF`,
```

Comment:
Asserting error content so that we verify that failure is for the wanted reason.

Does make tests more brittle to change in error message, but otherwise easy for test to "pass" even if failure is for an unexpected reason.